vortex-serde/src/layouts/read/mod.rs

```diff
@@ -58,7 +74,15 @@ pub trait LayoutReader: Debug + Send {
     /// and then call back into this function.
     ///
     /// The layout is finished reading when it returns None
-    fn read_next(&mut self) -> VortexResult<Option<ReadResult>>;
+    fn read_next(&mut self, selection: RowSelector) -> VortexResult<Option<ReadResult>>;
+
+    /// Produce sets of row ranges to read from underlying layouts.
+    ///
+    /// Range terminating at end of the layout indicates that layout is done producing ranges.
+    fn read_range(&mut self) -> VortexResult<Option<RangeResult>>;
+
+    /// Skip over next n_rows as we know they will not be part of result set
```

Comment:
I think this doc string is wrong? We're advancing up to row N, not skipping over N rows